File: katas/8 kyu/fake-binary/solution.py
import codewars_test as test
# -------------------------------- MY SOLUTION ------------------------------- #
def fake_bin(x):
  return ''.join(['1' if int(i) >= 5 else '0' for i in x])
# ------------------------------ CLEVER SOLUTION ----------------------------- #
def fake_bin(s):
  return s.translate(s.maketrans('0123456789', '0000011111'))
# ----------------------------------- TEST ----------------------------------- #
@test.describe("Fixed Tests")
def fixed_tests():
    @test.it('Basic Test Cases')
    def basic_test_cases():
        tests = [
            # [expected, input]
            ["01011110001100111", "45385593107843568"],
            ["101000111101101", "509321967506747"],
            ["011011110000101010000011011", "366058562030849490134388085"],
            ["01111100", "15889923"],
            ["100111001111", "800857237867"],
        ]
        
        for exp, inp in tests:
            test.assert_equals(fake_bin(inp), exp)

File: katas/8 kyu/fake-binary/solution.js
// fake-binary
/* ------------------------------- MY SOLUTION ------------------------------ */
function fakeBin(x){
  return x.split("").map(i => i < 5 ? '0' : '1').join("")
}
/* ----------------------------- CLEVER SOLUTION ---------------------------- */
function fakeBin(x) {
  return x.replace(/\d/g, d => d < 5 ? 0 : 1);
}
/* ---------------------------------- TEST ---------------------------------- */
// const chai = require("chai");
// const Test = chai.assert.strictEqual; // replace Test.assertEqual with Test
// chai.config.truncateThreshold=0;

const chai = require("chai");
const assert = chai.assert;
chai.config.truncateThreshold=0;

describe("Basic tests",() =>{
  it("Testing for fixed tests", () => {
    assert.strictEqual(fakeBin('45385593107843568'), '01011110001100111');
    assert.strictEqual(fakeBin('509321967506747'), '101000111101101'); 
    assert.strictEqual(fakeBin('366058562030849490134388085'), '011011110000101010000011011');
  })
});